docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,598 @@
+// Fetching posts List(1)
+// Fetching users List(2)
+
+//=> result: (Post, User) = (Post(1,2,An article with id 1),User(2,@egg_2))
+```
+
+In the previous example we fetched a post given its id, and then fetched its author. These
+data could come from entirely different places but Fetch makes working with heterogeneous sources
+of data very easy.
+
+### Parallelism
+
+Combining multiple independent requests to the same data source results in batching but what happens
+when combining independent requests to *different* data sources? We have the oportunity to query the
+different data sources at the same time.
+
+```scala
+val fch: Fetch[(Post, User)] = getPost(1).product(getUser(2))
+
+val result: (Post, User) = Fetch.run[Id](fch)
+// Fetching posts List(1)
+// Fetching users List(2)
+
+//=> result: (Post, User) = (Post(1,2,An article with id 1),User(2,@egg_2))
+```
+
+Since we are interpreting the fetch to the `Id` monad, that doesn't give us any concurrency, the fetches
+will be run sequentially. However, if we interpret it to a `Future` each request will run in its own logical
+thread.
+
+## Combinators
+
+Besides `flatMap` for sequencing fetches and `product` for running them concurrently, Fetch provides a number of
+other combinators.
+
+### Sequence
+
+Whenever we have a list of fetches of the same type and want to run them concurrently we can use the `sequence`
+combinator. It takes a `List[Fetch[A]]` and gives you back a `Fetch[List[A]]`, batching the fetches to the same
+data source and running fetches to different sources in parallel.
+
+Note that the `sequence` combinator is more general and not only works on lists but any type that has a [`Traverse`](http://typelevel.org/cats/tut/traverse.html) instance.
+
+```scala
+import cats.std.list._
+import cats.syntax.traverse._
+
+val fch: Fetch[List[User]] = List(getUser(1), getUser(2), getUser(3)).sequence
+
+val result: List[User] = Fetch.run[Id](fch)
+// Fetching users List(1, 2, 3)
+
+//=> result: List[User] = List(User(1,@egg_1), User(2,@egg_2), User(3,@egg_3))
+```
+
+As you can see, requests to the user data source were batched, thus fetching all the data in one round.
+
+### Traverse
+
+Another interesing combinator is `traverse`, which is the composition of `map` and `sequence`.
+
+```scala
+val fch: Fetch[List[User]] = List(1, 2, 3).traverse(getUser)
+
+val result: List[User] = Fetch.run[Id](fch)
+// Fetching users List(1, 2, 3)
+
+//=> result: List[User] = List(User(1,@egg_1), User(2,@egg_2), User(3,@egg_3))
+```
+
+# Interpreting a fetch to a concurrency monad
+
+Albeit the examples use `Id` as the concurrency Monad, `Fetch` is not limited to just `Id`, any monad `M` that
+implements `MonadError[M, Throwable]` will do. Fetch provides `MonadError` instances for some existing monads like
+`Future`, `cats.Id` and `cats.Eval` and is easy to write your own.
+
+In practice you'll be interpreting a fetch to a concurrency monad like `Future` or `scalaz.concurrent.Task` to exploit
```

Comment:
If we namespace Task we should namespace Future too as scalaz.concurrent also includes a lower primitive type called `Future` which is contained inside `Task`